contrib/intel/jenkins/tests.py

```diff
@@ -0,0 +1,309 @@
+        return True if (self.mpi == "impi") else False
+        
+    def execute_cmd(self):
+        command = self.cmd + self.options + self.imb_cmd
+        outputcmd = shlex.split(command)
+        common.run_command(outputcmd) 
+
+        
+class MpiTestStress(MpiTests):
+     
+    def __init__(self, branchname, buildno, testname, core_prov, fabric, 
+                 mpitype, hosts, ofi_build_mode, util_prov=None):
+        super().__init__(branchname, buildno, testname, core_prov, fabric, 
+                         mpitype,  hosts, ofi_build_mode, util_prov)
+        
+         
+        if((self.core_prov == "verbs" or self.core_prov =="psm2")):
+            self.n = 16
+            self.ppn = 8
+        else:
+            self.n = 4
+            self.ppn = 2
+      
+    @property
+    def stress_cmd(self):
+        return "{}/{}/stress/mpi_stress -dcr".format(self.libfab_installpath, self.mpi)
+
+    @property
+    def execute_condn(self):
+        return True if (self.mpi != 'ompi' or \
+                        self.ofi_build_mode != 'dbg') else  False
+    
+    def execute_cmd(self):
+        command = self.cmd + self.options + self.stress_cmd
+        outputcmd = shlex.split(command)
+        common.run_command(outputcmd) 
+
+         
+      
+class MpiTestOSU(MpiTests):
+
+    def __init__(self, branchname, buildno, testname, core_prov, fabric,
+                 mpitype, hosts, ofi_build_mode, util_prov=None):
+        super().__init__(branchname, buildno, testname, core_prov, fabric,
+                         mpitype, hosts, ofi_build_mode, util_prov)
+        
+        self.n = 4 
+        self.ppn = 2
+        self.two_proc_test_pattern = ['osu_latency',
+                                   'osu_bibw',
+                                   'osu_latency_mt',
+                                   'osu_bw','osu_get_latency',
+                                   'osu_fop_latency',
+                                   'osu_acc_latency',
+                                   'osu_get_bw',
+                                   'osu_put_latency',
+                                   'osu_put_bw',
+                                   'osu_put_bibw',
+                                   'osu_cas_latency',
+                                   'osu_get_acc_latency'
+            ]
+        self.osu_mpi_path = "{}/{}/osu/libexec/osu-micro-benchmarks/mpi/". \
+                            format(self.libfab_installpath,mpitype) 
+    
+    @property
+    def execute_condn(self): 
+        return True if (self.mpi != "ompi" or \
+                       (self.core_prov != "sockets" and \
+                        self.core_prov != "psm2" and \
+                        self.ofi_build_mode!="dbg")) \
+                    else False
+    
+    def execute_cmd(self):
+        assert(self.osu_mpi_path)
+        for root, dirs, files in os.walk(self.osu_mpi_path):
+            for file in files:
+                if file in self.two_proc_test_pattern:
```

Comment:
Rename file to test or benchmark. Rename `two_proc_test_pattern` to `two_proc_tests` and make it a dict.